docs/advanced_topics/customisation/custom_page_listings.md

```diff
@@ -0,0 +1,64 @@
+# Custom page listings
+
+Normally, editors navigate through the Wagtail admin interface by following the structure of the page tree. However, this can make it slow to locate a specific page for editing, especially on large sites where pages are organised into a deep hierarchy.
+
+Custom page listings are a way to present a flat list of all pages of a given type, accessed from a menu item in the Wagtail admin menu, with the ability for editors to search and filter this list to find the pages they are interested in.
+
+For example, if your site implemented the page type `BlogPage`, you could provide a "Blog pages" listing in the Wagtail admin by adding the following definitions to a `wagtail_hooks.py` file within the app:
+
+```python
+# myapp/wagtail_hooks.py
+from wagtail import hooks
+from wagtail.admin.viewsets.pages import PageListingViewSet
+
+from myapp.models import BlogPage
+
+
+class BlogPageListingViewSet(PageListingViewSet):
+    icon = "globe"
+    menu_label = "Blog Pages"
+    add_to_admin_menu = True
+    model = BlogPage
+
+
+blog_page_listing_viewset = BlogPageListingViewSet('blog_pages')
+@hooks.register('register_admin_viewset')
```

Comment:
For consistency and just in case we want to use blacken-docs in the future

```suggestion
blog_page_listing_viewset = BlogPageListingViewSet("blog_pages")
@hooks.register("register_admin_viewset")
```